packages/commonwealth/server/controllers/server_threads_methods/update_thread.ts

```diff
@@ -241,6 +227,19 @@ export async function __updateThread(
       { transaction },
     );
 
+    if (versionHistory) {
```

Comment:
From what I've gathered, the issue with detecting array change probably happens because Sequelize uses a shallow equality check– so if the reference of the array stays the same, it thinks nothing changed.

Creating a new array ref might fix the issue:
```ts
const newItems = [...entity.items]
// do stuff to newItems
entity.items = newItems
await entity.save()
```

Does this solve the issue in your case?